domain/src/main/java/org/oppia/domain/util/StateRetriever.kt

```diff
@@ -129,15 +132,24 @@ class StateRetriever @Inject constructor() {
       val inputKeysIterator = inputsJson.keys()
       while (inputKeysIterator.hasNext()) {
         val inputName = inputKeysIterator.next()
-        ruleSpecBuilder.putInput(inputName, createInputFromJson(inputsJson, inputName, interactionId))
+        when (ruleSpecBuilder.ruleType) {
+          "HasDenominatorEqualTo", "HasNumeratorEqualTo" -> ruleSpecBuilder.putInput(
```

Comment:
Is there a way to detect which InteractionObject to create based on the data rather than tying the parsing logic to specific rule types? This approach isn't very flexible, and our downstream code ensures that the types are correct during classification time.

Reply:
Unfortunately there is no easy way to do this since we still need to get the right json type. We cant really use the right get function without this method

Reply:
Ack. We'll need to figure out a way to do this cleanly in the long-term. Maybe having an interaction repository to look up the type would be an ideal way to share this information both with loaders and the classification system.

In the meantime, I guess we need to keep it this way.